check-compatibility-with-govuk-frontend-version.js

```diff
@@ -0,0 +1,65 @@
+// prototypes load and compile a subset of hmrc-frontend styles using their
+// own version of govuk-frontend, if their version of govuk-frontend is
+// lower than 5.4.0 then the compilation of styles will fail with an error
+// that can be hard for interaction designers to debug, we would like to
+// give them a warning on installation so that they don't get stuck later
+// when trying to use the hmrc-frontend styles.
+const fs = require('fs');
+const path = require('path');
+
+function hmrcFrontendDisableCompatibilityCheck() {
+  return process.env.HMRC_FRONTEND_DISABLE_COMPATIBILITY_CHECK === 'true';
+}
+
+function semverGreaterThanOrEqualTo(a, b) {
+  const [aMaj, aMin, aPat] = a.split('.').map(Number);
+  const [bMaj, bMin, bPat] = b.split('.').map(Number);
+  return ((aMaj - bMaj) || (aMin - bMin) || (aPat - bPat)) >= 0;
+}
+
+function withinPrototypeKitAndUsingIncompatibleGovukFrontendVersion(currentVersion) {
+  const minimumGovukFrontendVersion = '5.4.0';
+  return !semverGreaterThanOrEqualTo(currentVersion, minimumGovukFrontendVersion);
+}
+
+try {
+  const projectDir = process.env.INIT_CWD || process.cwd();
+  const projectPackagePath = path.join(projectDir, 'package.json');
+  const projectPackageJson = JSON.parse(fs.readFileSync(projectPackagePath, 'utf8'));
+
+  if (
+    !projectPackageJson.dependencies
+    || !projectPackageJson.dependencies['govuk-frontend']
+    || !projectPackageJson.dependencies['govuk-prototype-kit']
+  ) {
+    process.exit(0);
+  }
+
+  const currentVersionInProject = projectPackageJson.dependencies['govuk-frontend'].replace(/[^0-9.]/g, '');
+
```

Comment:
it looks good to me from a safety perspective, in that the projects this should impact are prototypes so it feels like a safe thing to introduce even though it will be a breaking change for them if they are using incompatible versions

from a taste perspective, I would move all this logic into withinPrototypeKitAndUsingIncompatibleGovukFrontendVersion and not pass any arguments to it - since it's just a script and we won't be testing parts of it but as a whole I don't think there's much value in having checks split out - where withinPrototypeKitAndUsingIncompatibleGovukFrontendVersion is a nice label to describe what that code is checking, but at the moment we're checking some of that stuff thats described in the function name outside of it, which I think is an easy footgun for future